packages/op-datasets/src/op_datasets/logic/transforms/daily_address_summary.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+import polars as pl
+
+
+CONDITIONS = {
+    "gas_price": lambda x: x > 0,
+}
+
+
+def daily_address_summary(
+    df: pl.DataFrame, conditions: dict[str, callable], **groupby_args: dict[str, str]
+) -> pl.DataFrame:
+    """Create a daily summary addresses and activities such as transaction count, gas usage, fees and more across the Superchain.
+
+    The main dataset we pull stats from is the `transactions` dataset.
+    We can filter the data based on the conditions set in the CONDITIONS dictionary.
+    The groupby_args dictionary is used to group the data by the columns specified in the dictionary.
+
+    In our pipeline, by default the function will group the following columns:
+    - `from_address`: the address that initiated a transaction
+    - `chain_id`: the chain id that the address
+    - `chain`: the chain name of a transaction
+    - `dt`: date granularity in the string format `YYYY-MM-DD`
+    """
+    # Create filter expression from conditions dictionary
+    filter_expr = pl.lit(True)
+    for col, condition in conditions.items():
+        filter_expr &= condition(pl.col(col))
+
+    # Apply the filter to the DataFrame
+    _filter_df = df.filter(filter_expr)
+
+    # convert timestamp from epoch to datetime
+    _filter_df = _filter_df.with_columns(
+        [
+            pl.from_epoch(pl.col("block_timestamp")),
+            pl.from_epoch(pl.col("block_timestamp"), time_unit="s")
+            .dt.hour()
+            .alias("hour"),
+        ]
+    )
+    # Extract groupby columns from the arguments
+    groupby_cols = list(groupby_args.values())
+
+    # SQL query to perform aggregations
+    groupby_clause = ", ".join(groupby_cols)
+
+    query = f"""
+    WITH constants AS (
+        SELECT
+            CAST(1000000000000000000 AS DECIMAL(18, 0)) AS WEI_IN_ETH,
+            CAST(1000000000 AS DECIMAL(9, 0)) AS WEI_IN_GWEI
+    )
+    ,df_enrich AS (
+        SELECT
+            _filter_df.*
+
+            -- Gas Fee Breakdown
+            ,CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS total_gas_fee
+            ,CASE
+                WHEN receipt_status = 1 THEN CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH
+                ELSE 0
+                END AS total_gas_fee_success
+            ,CASE
+                WHEN receipt_status != 1 THEN CAST(((gas_price * receipt_gas_used) + receipt_l1_fee) AS DECIMAL(38,18)) / constants.WEI_IN_ETH
+                ELSE 0
+                END AS total_gas_fee_fail
+            ,CAST((gas_price * receipt_gas_used) AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l2_contrib_gas_fee
+            ,CAST(receipt_l1_fee AS DECIMAL(38,18)) / constants.WEI_IN_ETH AS l1_contrib_gas_fee
```

Comment:
this should be equal to `l1_blobgas_contrib_gas_fee` + `l1_l1gas_contrib_gas_fee`

I think one of them needs to be divided by 16